File: gcj/y2018/practice/A.java
package gcj.y2018.practice;
import java.util.*;

public class A {
	public void run() {
		int low = in.nextInt();
		int high = in.nextInt() + 1;
		in.nextInt();
		for (;;) {
			int mid = (low + high) / 2;
			System.out.println(mid);
			System.out.flush();
			String response = in.next();
			switch (response) {
			case "CORRECT":
				return;
			case "TOO_SMALL":
				low = mid;
				break;
			case "TOO_BIG":
				high = mid;
				break;
			}
		}
	}
	
	static Scanner in;

	public static void main(String[] args) {
		in = new Scanner(System.in);
		int tests = in.nextInt();
		for (int t = 0; t < tests; t++) {
			new A().run();
		}
		in.close();
	}
}

File: gcj/y2018/practice/A.py
#!/usr/bin/python3
import collections


def solve():
	low, high = map(int, input().split())
	high += 1
	input()
	while True:
		mid = (low + high) // 2
		print(mid)
		resp = input()
		if resp == 'CORRECT':
			return
		if resp == 'TOO_BIG':
			high = mid
		else:
			low = mid


tests = int(input())
for t in range(tests):
	solve()
